ext/pg_connection.c

```diff
@@ -3904,10 +4006,9 @@ pgconn_set_default_encoding( VALUE self )
 
 	if (( enc = rb_default_internal_encoding() )) {
 		encname = pg_get_rb_encoding_as_pg_encoding( enc );
-		if ( pgconn_set_client_encoding_async(self, encname) != 0 )
+		if ( pgconn_set_client_encoding_async(self, rb_str_new_cstr(encname)) != 0 )
 			rb_warning( "Failed to set the default_internal encoding to %s: '%s'",
 			         encname, PQerrorMessage(conn) );
-		pgconn_set_internal_encoding_index( self );
```

Comment:
@larskanis  Sorry for resurrecting this ancient PR. But was removing `pgconn_set_internal_encoding_index` here intentional?

So this has the side effect of `internal_encoding` not being set correctly in the case the `set_client_encoding` call fails with an error for any reason.

This results in UTF-8 strings being interpreted as SQL_ASCII e.g
```
™™™™™™
becomes
\xE2\x84\xA2\xE2\x84\xA2\xE2\x84\xA2\xE2\x84\xA2\xE2\x84\xA2\xE2\x84\xA2
```
This not the case in version 1.2.3